### PR TITLE
Pricing-modal-display_CU-nx9tj3

### PR DESCRIPTION
**Pull request by a-wilkes**

### CODE REVIEW

**Comment by a-wilkes** [comment]
Task linked: [CU-nx9tj3 Pricing modal display](https://app.clickup.com/t/nx9tj3)

**Comment by lpboucher** [comment]
Task linked: [CU-nx9tj3 Pricing modal display](https://app.clickup.com/t/nx9tj3)

**Comment by lpboucher** [comment]
Task linked: [CU-nx9t53 Price discount modal](https://app.clickup.com/t/nx9t53)

File: src/app/modules/backbone/pages/backbone/backbone.page.ts [hunk 1]
```diff
@@ -1,14 +1,42 @@
...
-export class BackbonePageComponent {
+export class BackbonePageComponent implements OnDestroy {
     showSidebar: boolean = false;
+    private subscriptions: Subscription = new Subscription();
+
+    constructor(
+        private activatedRoute: ActivatedRoute,
+        private router: Router,
+        private modalService: ModalService<VolumeSelectionModalComponent>,
+    ) {
```
Comment (lpboucher) [inline src/app/modules/backbone/pages/backbone/backbone.page.ts]: subscriptions should be added to ngOnInit

File: src/app/modules/backbone/pages/backbone/backbone.page.ts [hunk 2]
```diff
@@ -1,14 +1,42 @@
...
+        this.subscriptions.add(
+            this.activatedRoute.queryParams.subscribe(
```
Comment (lpboucher) [inline src/app/modules/backbone/pages/backbone/backbone.page.ts]: if done on component init and not on subsequent changes, can use snapshot.queryParams

File: src/app/modules/backbone/pages/backbone/backbone.page.ts [hunk 3]
```diff
@@ -1,14 +1,42 @@
...
+                (params: Params) => {
+                    if (params["checkPricing"] === "true") {
+                        this.modalService.open(VolumeSelectionModalComponent);
 
-    constructor() {}
+                        const { checkPricing, ...newParams } = params;
```
Comment (lpboucher) [inline src/app/modules/backbone/pages/backbone/backbone.page.ts]: you can destructure this variable in the callback directly

File: src/app/modules/backbone/pages/backbone/backbone.page.ts [hunk 4]
```diff
@@ -1,14 +1,42 @@
...
+                        this.router.navigate([], { queryParams: newParams });
```
Comment (lpboucher) [inline src/app/modules/backbone/pages/backbone/backbone.page.ts]: this logic should be done within the modal onConfirm method

File: src/app/modules/backbone/pages/backbone/backbone.page.ts [hunk 5]
```diff
@@ -1,14 +1,42 @@
...
 
 @Component({
     selector: "app-backbone",
     templateUrl: "./backbone.page.html",
     styleUrls: ["./backbone.page.scss"]
 })
-export class BackbonePageComponent {
+export class BackbonePageComponent implements OnDestroy {
     showSidebar: boolean = false;
+    private subscriptions: Subscription = new Subscription();
```
Comment (lpboucher) [inline src/app/modules/backbone/pages/backbone/backbone.page.ts]: private properties should be listed above public ones

Comment (lpboucher) [reply src/app/modules/backbone/pages/backbone/backbone.page.ts]: removed subscription

File: src/app/modules/backbone/pages/backbone/backbone.page.ts (hunk 1 above)
Comment (lpboucher) [reply src/app/modules/backbone/pages/backbone/backbone.page.ts]: used snapshot in OnInit

File: src/app/modules/backbone/pages/backbone/backbone.page.ts (hunk 2 above)
Comment (lpboucher) [reply src/app/modules/backbone/pages/backbone/backbone.page.ts]: done

File: src/app/modules/backbone/pages/backbone/backbone.page.ts (hunk 3 above)
Comment (lpboucher) [reply src/app/modules/backbone/pages/backbone/backbone.page.ts]: done

File: src/app/modules/backbone/pages/backbone/backbone.page.ts (hunk 4 above)
Comment (lpboucher) [reply src/app/modules/backbone/pages/backbone/backbone.page.ts]: done

**Review by lpboucher — APPROVED** [review]

**Pull request merged**